### PR TITLE
feat: [CDS-66804]: Setting for default gitExperience repo

### PR DESCRIPTION
**Pull request by sunnykesh-harness**
### Summary
**Feature summary** 
A new GItX setting for default repository is defined. So, if user has set any repository as default, it should get autofilled while doing remote operations. 

**Changes are : :** 

- A new settingComponent for Text Input is defined.
- Changes are done in common component of `GitSync` form so all applicable use cases will be handled whenever no repository was selected by default. `Test added` .
- These setting API should not be called in edit mode where we already have connector, repo etc.
- **Edge case handled** : User has not added default repo in list of allowed repo then repository should not be pre-filled. `Test added`
- In gitSync form now two of the settings are used so switching to settingList API. Before, API to fetch single setting ( default connector) was used.
- Fixed exiting tests API mocks

#### Screenshots
**Default repository setting**

Screenshot 2023-08-01 at 1 22 14 AM

**Auto filling repository from the setting**

https://github.com/harness/harness-core-ui/assets/63278928/15dde579-1260-46a9-a2e9-fb3f0133e46a

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/27-platform/default-settings/components/__test__/ReusableHandlers.test.tsx [hunk 1]
```diff
@@ -102,7 +102,7 @@ describe('Reusable Components', () => {
     const { container } = renderObj
     expect(container).toMatchSnapshot()
   })
-  test('DefaultSettingTextbox', () => {
+  test('DefaultSettingMultiTextbox', () => {
```
Comment (rameshx) [inline src/modules/27-platform/default-settings/components/__test__/ReusableHandlers.test.tsx:105]: Looks like `DefaultSettingNumberTextbox` is used in the test
```suggestion
  test('DefaultSettingNumberTextbox', () => {
```

**Review by rameshx — DISMISSED** [review]
One minor comment that can be addressed later

File: src/modules/27-platform/default-settings/components/__test__/ReusableHandlers.test.tsx (hunk 1 above)
Comment (sunnykesh-harness) [reply src/modules/27-platform/default-settings/components/__test__/ReusableHandlers.test.tsx]: yes @rameshx . Corrected test title.

**Review by avinashmadhwani02 — APPROVED** [review]

**Review by rameshx — APPROVED** [review]
